src/App.tsx

```diff
@@ -15,49 +20,94 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+function getSortedGoods(
+  goods: ProductList,
+  { appliedSort, reverseOrder }: FilteredGoods,
+): string[] {
+  const visibleGoods = [...goods];
+
+  if (appliedSort) {
+    visibleGoods.sort((good1, good2) => {
+      switch (appliedSort) {
+        case SortVariants.APPLIED_SORT_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SortVariants.APPLIED_SORT_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseOrder) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [appliedSort, setAppliedSort] = useState('');
+  const [reverseOrder, setReverseOrder] = useState(false);
+
+  const reset = () => {
+    setAppliedSort('');
+    setReverseOrder(false);
+  };
+
+  const visibleGoods:ProductList = getSortedGoods(goodsFromServer,
```

Comment:
it's better to wrap the result of function inside useMemo hook to avoid extra re-rendering